subsys/bluetooth/mesh/friend.c

```diff
@@ -52,17 +50,18 @@ struct friend_pdu_info {
 };
 
 NET_BUF_POOL_FIXED_DEFINE(friend_buf_pool, FRIEND_BUF_COUNT,
-			  BT_MESH_ADV_DATA_SIZE, bt_mesh_adv_buf_destroy);
+			  BT_MESH_ADV_DATA_SIZE, NULL);
 
 static struct friend_adv {
-	struct bt_mesh_adv adv;
 	uint16_t app_idx;
```

Comment:
Another way is to retain the friend destructor in friend, through friend_pool allocates a buffer, but data deep copy is still required,  and this part of memory cannot be saved, so it is not desirable.
I think the default configuration can save `sizeof(struct bt_mesh_adv) * ((CONFIG_BT_MESH_FRIEND_QUEUE_SIZE + 1) * CONFIG_BT_MESH_FRIEND_LPN_COUNT)` = 12 * ((16 + 1) * 2) = 408bytes  of memory unexpected.